### DISCUSSION TITLE
Discuss about a possible `release-changelog` action

### DISCUSSION THREAD

**Discussion by jorgepiloto** [pyansys/actions#234] [Ideas]
Hi all, I wanted to get your feedback on a design decision about an automated changelog for our libraries.

**Background 📝**
The idea is to use https://towncrier.readthedocs.io/en/stable/. This tool is based on the idea of "fragments". Everytime you make a change in your code you need to write a small RST file. This file is named <PR number>.<type of change>. For example:

```
1234.bugfix
2345.feature
```

When you are ready to release, you can execute "towncrier build <version>". The tool glues all the fragments and builds a new entry for the changelog.

**The issue 🐞**
Let's say that a developer performs a release using an action named `release-changelog`. The actions runs the `towncrier build ...` command and the desired changelog file gets updated.

These new changes are new, so a new tag is required.

This introduces a chicken-egg problem: tag -> update CHANGELOG -> new changes -> commit changes -> tag

**Possible solutions 💡**

1. Create a `changelog-style` action that checks that the CHANGELOG contains the tag that triggered the workflow. This forces developers to run locally the `towncrier build ...` action, manually commit the updated changelog, pushing and finally tagging before the release.

**Comment by RobPasMue** [comment]
It is not possible that we are the first to experience this issue... I'm looking at their docs and I might fail to see the problem here. I'd appreciate some kind of pair-programming session @jorgepiloto - I'd be happy to help.
